Knoema.Client/Client.cs

```diff
@@ -117,12 +117,18 @@ private Task<T> ApiPost<T>(string path, object obj)
 
 		public Task<IEnumerable<Dataset>> ListDatasets(string source = null, string topic = null, string region = null)
 		{
-			var query = string.Empty
-				.AddUrlParam("source", source)
-				.AddUrlParam("topic", topic)
-				.AddUrlParam("region", region);
 
-			return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset", query);
+			if (string.IsNullOrEmpty(source ?? topic ?? region))
+				return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset");
```

Comment:
1. Different endpoints used as one endpoint does simple listing of datasets but its output can get slow sometimes. Other endpoint allows filtering and is faster when datasets needed on filtering criteria example by source.
2. Currently the properties in Dataset class here were sufficient with current requirements.